StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/SpacingRules/SA1023CSharp9UnitTests.cs

```diff
@@ -3,9 +3,66 @@
 
 namespace StyleCop.Analyzers.Test.CSharp9.SpacingRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Testing;
     using StyleCop.Analyzers.Test.CSharp8.SpacingRules;
+    using Xunit;
+    using static StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly,
+        StyleCop.Analyzers.SpacingRules.TokenSpacingCodeFixProvider>;
 
     public class SA1023CSharp9UnitTests : SA1023CSharp8UnitTests
     {
+        [Fact]
+        public async Task TestFunctionPointerParameterValidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidPrecedingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int *> FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotPreceded).WithLocation(3, 26);
```

Comment:
💡  Can you switch these tests to use the `WithLocation(0)` form? The `testCode` will contain matching syntax like `{|#0: |}` to indicate the location of diagnostic 0 within the input.